app/grandchallenge/core/utils/markdown.py

```diff
@@ -9,22 +12,64 @@ def extendMarkdown(self, md):  # noqa: N802
 
 
 class BS4Treeprocessor(Treeprocessor):
+
     def run(self, root):
+
+        el_class_dict = {
+            "img": "img-fluid",
+            "blockquote": "blockquote",
+            "table": "table table-hover table-borderless",
+            "thead": "thead-light",
+            "code": "codehilite",
+        }
+
         for el in root.iter():
-            if el.tag == "img":
-                el.set("class", "img-fluid")
 
-            elif el.tag == "blockquote":
-                el.set("class", "blockquote")
+            if el.tag in el_class_dict:
+                BS4Treeprocessor.set_css_class(el, el_class_dict[el.tag])
+
+        for i in range(len(self.md.htmlStash.rawHtmlBlocks)):
+
+            bs4block = BeautifulSoup(
+                self.md.htmlStash.rawHtmlBlocks[i], "html.parser"
+            )
+
+            el = bs4block.find()
```

Comment:
What is this finding? Everything? Why not only the relevant things you're looking for?

Reply:
We have a dictionary of elements to add classes two, and a list of rawHtmlBlocks. So either loop over the elements in the HtmlBlock and check if the element is in the dictionary, or, loop over the dictionary and `find_all` the element in the HtmlBlock. The dictionary is not long (for now) and also the HtmlBlocks not expected to be long (since the editor is a markdown editor and raw html is not the norm there), so I don't have a hard opinion here. However, I followed your comment and checked for each element in the dictionary with `find_all()`.